python/cugraph/tests/test_betweenness_centrality.py

```diff
@@ -30,82 +32,448 @@
+    assert isinstance(k, int), "This test is meant for verifying coherence " \
+                               "when k is given as an int"
+    # In the fixed set we compare cu_bc against istelf as we random.seed(seed)
+    # on the same seed and then sample on the number of vertices themselves
+    random.seed(seed)  # It will be called again in nx's call
+    sources = random.sample(range(G.number_of_vertices()), k)
+    # The first call is going to proceed to the random sampling in the same
+    # fashion as the lines above
+    df = cugraph.betweenness_centrality(G, normalized=normalized,
+                                        weight=weight,
+                                        endpoints=endpoints,
+                                        k=k,
+                                        seed=seed,
+                                        implementation=implementation,
+                                        result_dtype=result_dtype)
 
+    # The second call is going to process source that were already sampled
+    # We set seed to None as k : int, seed : not none should not be normal
+    # behavior
+    df2 = cugraph.betweenness_centrality(G, normalized=normalized,
+                                         weight=weight,
+                                         endpoints=endpoints,
+                                         k=sources,
+                                         seed=None,
+                                         implementation=implementation,
+                                         result_dtype=result_dtype)
+    cu_bc = {key: score for key, score in
+             zip(df['vertex'].to_array(),
+                 df['betweenness_centrality'].to_array())}
+    cu_bc2 = {key: score for key, score in
+              zip(df2['vertex'].to_array(),
+                  df2['betweenness_centrality'].to_array())}
+
+    return cu_bc, cu_bc2
+
+
+def _calc_bc_full(G, Gnx, normalized, weight, endpoints, implementation,
+                  k, seed,
+                  result_dtype):
+    df = cugraph.betweenness_centrality(G, normalized=normalized,
+                                        weight=weight,
+                                        endpoints=endpoints,
+                                        implementation=implementation,
+                                        result_dtype=result_dtype)
+    nx_bc = nx.betweenness_centrality(Gnx, normalized=normalized,
+                                      weight=weight,
+                                      endpoints=endpoints)
+
+    cu_bc = {key: score for key, score in
+             zip(df['vertex'].to_array(),
+                 df['betweenness_centrality'].to_array())}
+    return cu_bc, nx_bc
+
+
+# =============================================================================
+# Utils
+# =============================================================================
+def prepare_rmm(managed_memory, pool_allocator):
+    gc.collect()
     rmm.reinitialize(
```

Comment:
This won't fly with #849.
Removing some combinations will help to reduce the runtime of all the BC tests

Reply:
Updated in bc82d30, `prepare_rmm()` is replaced by `prepare_test()` that only calls `gc.collect()` for now.
Tests are no longer parametrized with rmm parameters